.github/workflows/publish-major-minor.yml

```
@@ -0,0 +1,33 @@
+name: "Publish Major-Minor-Tags"
+on:
+  push:
+    tags:
+      - "v*"
+
+jobs:
+  push-tags:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install packaging
+        run: pip install -U packaging
+      - name: Get versions
+        id: get_versions
+        run: python .github/workflows/publish-major-minor.py ${{ github.ref }}
+      - name: Push Tags Version
+        if: steps.get_versions.outputs.original_tag_name != ''
+        env:
+          original_tag_name: ${{ steps.get_versions.outputs.original_tag_name }}
+          major_version: ${{ steps.get_versions.outputs.major_version }}
+          minor_version: ${{ steps.get_versions.outputs.minor_version }}
+        run: |
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git config --global user.name "github-actions[bot]"
+          git tag -a $major_version -m "original tag: $original_tag_name"
+          git tag -a $minor_version -m "original tag: $original_tag_name"
+          git push origin $major_version -f
+          git push origin $minor_version -f
```

It's better to do `--force-with-lease` to avoid race conditions. And if we use GHA-based interpolation, we can use single quotes in bash because the contents would be pre-rendered
```suggestion
          git push origin '${{ env.major_version }}' --force-with-lease
          git push origin '${{ env.minor_version }}' --force-with-lease
```

---

I didn't know that flag until now, always nice to learn new stuff 😄 
But it looks like, [it isn't playing nicely with tags](https://github.com/s-weigand/gh-action-pypi-publish/runs/1350418417?check_suite_focus=true)
Also since the original tag is preserved and pushing tags is quite rare, IMHO there is a very low chance for race conditions and even if you should always be able to recover.
```console
$ git push --delete origin <tagname>
$ git push origin <tagname>
```

---

Noted, thanks for testing :)